tests/Baseline/spicy.optimization.default-parser-functions/log

```diff
@@ -1,4 +1,21 @@
 ### BTest baseline data generated by btest-diff. Do not edit. Use "btest -U/-u" to update. Requires BTest >= 0.63.
+[debug/global-optimizer] disabling feature 'supports_sinks' of type 'foo::P1' since it is not used
+[debug/global-optimizer] disabling feature 'supports_sinks' of type 'foo::P2' since it is not used
+[debug/global-optimizer] disabling feature 'uses_random_access' of type 'foo::P2' since it is not used
+[debug/global-optimizer] inlining constant 'foo::__feat%foo__P1%supports_filters' at ''
+[debug/global-optimizer] inlining constant 'foo::__feat%foo__P1%supports_filters' at ''
+[debug/global-optimizer] inlining constant 'foo::__feat%foo__P1%supports_filters' at ''
+[debug/global-optimizer] inlining constant 'foo::__feat%foo__P2%supports_filters' at ''
+[debug/global-optimizer] inlining constant 'foo::__feat%foo__P2%supports_filters' at ''
+[debug/global-optimizer] inlining constant 'foo::__feat%foo__P2%supports_filters' at ''
+[debug/global-optimizer] removing conditional code '' which can never be reached
+[debug/global-optimizer] removing conditional code '' which can never be reached
+[debug/global-optimizer] removing conditional code around '' since the condition is always true
```

Comment:
We should add some dummy location to the conditional, feature-dependent code we generate in the parser builder so we can log something meaningful here.